plugins/inputs/mavlink/README.md

```diff
@@ -0,0 +1,129 @@
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Read metrics from a Mavlink connection to a flight controller.
+[[inputs.mavlink]]
+     # Flight controller URL. Must be a valid Mavlink connection string in one
+     # of the following formats:
+     #
+     # - Serial port:  serial://<device name>:<baud rate> 
+     #            eg: "serial:///dev/ttyACM0:57600"
+     # 
+     # - TCP client:   tcp://<target ip or hostname>:<port>
+     #            eg: "tcp://192.168.1.12:14550"
+     # 
+     # - UDP client:   udp://<target ip or hostname>:<port>
+     #            eg: "udp://192.168.1.12:14550"
+     # 
+     # - UDP server:   udp://:<listen port>
+     #            eg: "udp://:14540"
+     # 
+     # The meaning of each of these modes is documented by
+     # https://mavsdk.mavlink.io/v1.4/en/cpp/guide/connections.html.
+     fcu_url = "udp://:14540"
+
+     # Filter to specific messages. Only the messages in this list will be parsed.
+     # If blank, all messages will be accepted.
+     message_filter = []
+
+     # Mavlink system ID for Telegraf
+     # Only used if the mavlink plugin is sending messages, eg.
+     # when `stream_request_enable` is enabled (see below.)
+     system_id = 254
+
+     # Determines whether the plugin sends requests to stream telemetry,
+     # and if enabled, the requested frequency of telemetry in Hz.
+     # This setting should be disabled if your software controls rates using
+     # REQUEST_DATA_STREAM or MAV_CMD_SET_MESSAGE_INTERVAL
+     # (See https://mavlink.io/en/mavgen_python/howto_requestmessages.html#how-to-request--stream-messages)
+     stream_request_enable = true
+     stream_request_frequency = 4
+```
+
+### Note: Mavlink Dialects
+
+This plugin currently only uses the ArduPilot-specific dialect, which also
+includes messages from the common Mavlink dialect.
+
+See the [Mavlink docs](https://mavlink.io/en/messages/ardupilotmega.html) for
+more info on dialects.
+
+## Metrics
+
+Each supported Mavlink message translates to one metric group, and fields
+on the Mavlink message are converted to fields in telegraf.
+
+The name of the Mavlink message is translated into lowercase and any
+leading text `message_` is dropped.
+
+For example, [MESSAGE_ATTITUDE](https://mavlink.io/en/messages/common.html)
+will become an `attitude` metric, with all fields copied from its Mavlink
+message definition.
+
+## Example Output
+
+_`mavlink` input plugin connected to ArduPilot SITL and the `file` output
+plugin:_
+
+```text
+system_time,host=chris-ubuntu time_boot_ms=320734i,time_unix_usec=0i 1732249197549336855
```

Comment:
The code shows a tag named `fcu_url` and `sys_id` to be always added, I don't see this reflected here.

Note that a convention is to also have a `source` tag to indicate from which external system a metric is coming from.

PS: You can remove the `host=..` tag, since that will be dependent on user settings and is irrelevant for the sample metrics.

Reply:
Good catch, this was generated from an older build, updated this now so it uses the latest build and removed the `host` tags!